x-pack/plugins/apm/server/lib/services/get_service_dependencies/index.ts

```diff
@@ -43,187 +49,254 @@ export type ServiceDependencyItem = {
+      type: 'external' as const,
+      spanType: item.span.type,
+      spanSubtype: item.span.subtype,
+      ...destMetrics,
+    };
+  });
+
+  const latencySums = metricsByResolvedAddress
+    .map(
+      (metric) => (metric.latency.value ?? 0) * (metric.throughput.value ?? 0)
+    )
+    .filter(isFiniteNumber);
+
+  const minLatencySum = Math.min(...latencySums);
+  const maxLatencySum = Math.max(...latencySums);
+
+  return metricsByResolvedAddress.map((metric) => {
+    const impact =
+      isFiniteNumber(metric.latency.value) &&
+      isFiniteNumber(metric.throughput.value)
+        ? ((metric.latency.value * metric.throughput.value - minLatencySum) /
+            (maxLatencySum - minLatencySum)) *
+          100
+        : 0;
+
+    return {
+      ...metric,
+      impact,
+    };
+  });
+}
+
 export function getServiceDependencies({
   setup,
   serviceName,
   environment,
   numBuckets,
+  comparisonStart,
+  comparisonEnd,
 }: {
   serviceName: string;
   setup: Setup & SetupTimeRange;
   environment?: string;
   numBuckets: number;
+  comparisonStart: number;
+  comparisonEnd: number;
 }): Promise<ServiceDependencyItem[]> {
   return withApmSpan('get_service_dependencies', async () => {
     const { start, end } = setup;
-    const [allMetrics, destinationMap] = await Promise.all([
+    const commonProps = {
+      setup,
+      serviceName,
+      environment,
+      numBuckets,
+    };
+    const [
+      currentPeriodMetrics,
+      previousPeriodMetrics,
+      destinationMap,
+    ] = await Promise.all([
+      getMetrics({ ...commonProps, start, end }),
       getMetrics({
-        setup,
-        serviceName,
-        environment,
-        numBuckets,
-      }),
-      getDestinationMap({
-        setup,
-        serviceName,
-        environment,
+        ...commonProps,
+        start: comparisonStart,
+        end: comparisonEnd,
       }),
+      getDestinationMap({ setup, serviceName, environment }),
     ]);
 
-    const metricsWithDestinationIds = allMetrics.map((metricItem) => {
-      const spanDestination = metricItem.span.destination.service.resource;
-
-      const destination = maybe(destinationMap[spanDestination]);
-      const id = destination?.id || {
-        [SPAN_DESTINATION_SERVICE_RESOURCE]: spanDestination,
-      };
-
-      return merge(
-        {
-          id,
-          metrics: [metricItem],
-          span: {
-            destination: {
-              service: {
-                resource: spanDestination,
-              },
-            },
-          },
-        },
-        destination
-      );
-    }, []);
-
-    const metricsJoinedByDestinationId = joinByKey(
-      metricsWithDestinationIds,
-      'id',
-      (a, b) => {
-        const { metrics: metricsA, ...itemA } = a;
-        const { metrics: metricsB, ...itemB } = b;
-
-        return merge({}, itemA, itemB, { metrics: metricsA.concat(metricsB) });
-      }
+    const currentPeriod = calculateMetrics(
```

Comment:
can you use an object here rather than different arguments? that makes it easier to catch mistakes (e.g., you might swap start and end), and easier to change. Usually anything beyond 2 arguments is better off as an object.